routers/web/repo/pull.go

```diff
@@ -1318,6 +1318,15 @@ func CleanUpPullRequest(ctx *context.Context) {
 
 func deleteBranch(ctx *context.Context, pr *models.PullRequest, gitRepo *git.Repository) {
 	fullBranchName := pr.HeadRepo.Owner.Name + "/" + pr.HeadBranch
+
+	if setting.Repository.PullRequest.RetargetChildsOnClose && pr.BaseRepoID == pr.HeadRepoID {
+		if err := pull_service.RebaseBranchPulls(ctx, ctx.Doer, pr.HeadRepoID, pr.HeadBranch, pr.BaseBranch); err != nil {
+			log.Error("RebaseBranchPulls: %v", err)
+			ctx.Flash.Error(ctx.Tr("repo.branch.deletion_failed", fullBranchName))
+			return
+		}
+	}
+
 	if err := repo_service.DeleteBranch(ctx.Doer, pr.HeadRepo, gitRepo, pr.HeadBranch); err != nil {
```

Comment:
Why define this block twice when you can also move it into the mentioned `repo_service.DeleteBranch` and enhance the error handling?

Reply:
Good idea - will do it.

Reply:
I've made error handling a little bit better, but it's not possible to move the whole code to `repo_service.DeleteBranch`.